dispatcher/main.go

```diff
@@ -0,0 +1,480 @@
+	info(fmt.Sprintf("To schedule %d tests.", len(settings)))
+	err = nil
+	tasks := make([]models.Task, len(settings))
+
+	for idx, setting := range settings {
+		var task models.Task
+		task.Name = fmt.Sprintf("Test: %s", setting.Classifier["identifier"])
+		task.Settings = setting
+		task.Annotation = run.Settings[common.KeyImageName].(string)
+
+		tasks[idx] = task
+	}
+
+	templateError := fmt.Sprintf("Fail to create task for run %d.", run.ID) + " Reason %s. Exception %s."
+	info("Posting tasks ...")
+	_, err = sendRequest("POST", fmt.Sprintf("run/%d/tasks", run.ID), tasks, templateError)
+	info("Finish posting tasks ...")
+
+	return
+}
+
+func createTaskJob(run *models.Run) (job *batchv1.Job, err error) {
+	client, err := kubeutils.CreateKubeClientset()
+	if err != nil {
+		return nil, err
+	}
+
+	parallelism := int32(run.Settings[common.KeyInitParallelism].(float64))
+	var backoff int32 = 5
+
+	name := fmt.Sprintf("%s-%d-%s", droidMetadata.Product, run.ID, getRandomString())
+	definition := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: batchv1.JobSpec{
+			Parallelism:  &parallelism,
+			BackoffLimit: &backoff,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+				Spec: corev1.PodSpec{
+					Containers:       getContainerSpecs(run),
+					ImagePullSecrets: getImagePullSource(run),
+					Volumes:          getVolumes(run),
+					RestartPolicy:    corev1.RestartPolicyNever,
+				},
+			},
+		},
+	}
+
+	return client.BatchV1().Jobs(namespace).Create(&definition)
+}
+
+func getLabels(run *models.Run) map[string]string {
+	labels := make(map[string]string)
+	labels["run_id"] = strconv.Itoa(run.ID)
+	labels["run_live"] = run.Settings[common.KeyLiveMode].(string)
+
+	return labels
+}
+
+func getVolumes(run *models.Run) []corev1.Volume {
+	if droidMetadata.Storage {
+		azureSource := corev1.AzureFileVolumeSource{
+			SecretName: getSecretName(run),
+			ShareName:  run.Settings[common.KeyStorageShare].(string),
+		}
+		return []corev1.Volume{
+			corev1.Volume{
+				Name:         common.NameStorageVolume,
+				VolumeSource: corev1.VolumeSource{AzureFile: &azureSource}}}
+	}
+
+	return []corev1.Volume{}
+}
+
+func getImagePullSource(run *models.Run) []corev1.LocalObjectReference {
+	return []corev1.LocalObjectReference{corev1.LocalObjectReference{Name: run.Settings[common.KeyImagePullSecret].(string)}}
+}
+
+func getContainerSpecs(run *models.Run) []corev1.Container {
+	c := corev1.Container{
+		Name:  "main",
+		Image: run.Settings[common.KeyImageName].(string),
+		Env:   getEnvironmentVariableDef(run)}
+
+	if droidMetadata.Storage {
+		c.VolumeMounts = []corev1.VolumeMount{corev1.VolumeMount{
+			MountPath: common.PathMountStorage,
+			Name:      common.NameStorageVolume}}
+	}
+
+	return []corev1.Container{c}
+}
+
+func getEnvironmentVariableDef(run *models.Run) []corev1.EnvVar {
+	result := []corev1.EnvVar{
+		corev1.EnvVar{
+			Name:      "ENV_POD_NAME",
```

Comment:
>ENV_POD_NAME [](start = 15, length = 12)

Some of these are already in the `common` package

Reply:
Yup!